File: backend/app/schemas/notification.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import NotificationType


class NotificationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    tenant_id: uuid.UUID
    store_id: uuid.UUID
    type: NotificationType
    title: str
    body: str
    is_read: bool
    created_for_user_id: uuid.UUID | None
    reference_id: uuid.UUID | None
    created_at: datetime

File: storemate-tn/frontend/src/types/notification.ts
export type NotificationType = "low_stock" | "subscription" | "system";

export interface Notification {
  id: string;
  tenant_id: string;
  store_id: string;
  type: NotificationType;
  title: string;
  body: string;
  is_read: boolean;
  created_for_user_id: string | null;
  reference_id: string | null;
  created_at: string;
}
